cmd/frontend/internal/app/verify_email.go

```diff
@@ -46,6 +45,19 @@ func serveVerifyEmail(db dbutil.DB) func(w http.ResponseWriter, r *http.Request)
 			return
 		}
 		verified, err := database.UserEmails(db).Verify(ctx, usr.ID, email, verifyCode)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("User %d email %q cannot be verified", usr.ID, email), http.StatusBadRequest)
+			return
+		}
+		// Set the verified email as primary if user has no existing email
+		emails, err := database.GlobalUserEmails.ListByUser(ctx, database.UserEmailsListOptions{
```

Comment:
```suggestion
		emails, err := database.UserEmails(db).ListByUser(ctx, database.UserEmailsListOptions{
```

This is better and the same, as we're trying to deprecate the old `Global*` variant in favor of explicitly passing the `db` around.

Reply:
That's good to know! Thank you @slimsag!